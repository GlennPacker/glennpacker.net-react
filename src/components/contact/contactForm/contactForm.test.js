import React from 'react';
import { shallow } from 'enzyme';

import ContactForm from './index';

describe('Contact Form', () => {
    let wrapper;
    const mockSubmit = jest.fn();
    const mockFieldChange = jest.fn();

    beforeAll(() => {
        const contactDetails = {
            name: 'john',
            email: "email",
            message: "message"
        }



        wrapper = shallow(
            <ContactForm
                contactDetails={contactDetails}
                submitForm={mockSubmit}
                onFieldChange={mockFieldChange}
                sending={false}
            />
        );
    })

    it('should render matching snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })
})