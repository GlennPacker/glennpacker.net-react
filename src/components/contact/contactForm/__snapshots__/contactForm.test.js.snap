// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Form should render matching snapshot 1`] = `
<div
  className="container pt-3"
>
  <form
    onSubmit={[MockFunction]}
  >
    <div
      className="form-group"
    >
      <label
        htmlFor="name"
      >
        Name
      </label>
      <input
        className="form-control"
        name="name"
        onChange={[MockFunction]}
        placeholder="Enter name"
        required={true}
        type="text"
        value="john"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="email"
      >
        Email address
      </label>
      <input
        className="form-control"
        name="email"
        onChange={[MockFunction]}
        placeholder="Enter email"
        type="email"
        value="email"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="message"
      >
        Message
      </label>
      <textarea
        className="form-control"
        name="message"
        onChange={[MockFunction]}
        placeholder="Enter message"
        required={true}
        rows="3"
        value="message"
      />
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Send
    </button>
  </form>
</div>
`;
